react/features/whiteboard/functions.ts

```diff
@@ -0,0 +1,120 @@
+import md5 from 'js-md5';
+
+// @ts-ignore
+import { getPinnedParticipant } from '../../features/base/participants';
+import { getRemoteParticipants, isLocalParticipantModerator } from '../base/participants/functions';
+import { appendURLParam } from '../base/util/uri';
+
+// @ts-ignore
+import { getCurrentRoomId, isInBreakoutRoom } from '../breakout-rooms/functions';
+
+// @ts-ignore
+import { _electParticipantInLargeVideo } from '../large-video/actions.any';
+import { hideWhiteboard, showWhiteboard, openWhiteboard } from './actions';
+import { WHITEBOARD_ID } from './constants';
+
+const getWhiteboardState = (state: any): any => state['features/whiteboard'];
+
+// TODO: remove negation
+/**
+ * Indicates whether the whiteboard is enabled in the config.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardEnabled = (state: any): boolean => !state['features/base/config'].whiteboard?.enabled;
+
+/**
+ * Indicates whether the whiteboard is open.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardOpen = (state: any): boolean => getWhiteboardState(state).isOpen;
+
+/**
+ * Indicates whether the whiteboard is present as a meeting participant.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardPresent = (state: any): boolean => getRemoteParticipants(state).has(WHITEBOARD_ID);
+
+/**
+ * Returns the whiteboard collaboration link.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {{ roomId: string, roomKey: string}|null}
+ */
+export const getCollabDetails = (state: any): {
+    roomId: string; roomKey: string;
+} => getWhiteboardState(state).collabDetails || {};
+
+/**
+ * Returns the whiteboard collaboration server url.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {string}
+ */
+export const getCollabServerUrl = (state: any): string => {
+    const collabServerBaseUrl = state['features/base/config'].whiteboard?.collabServerBaseUrl
+    || 'https://excalidraw-backend-pilot.jitsi.net'; // TODO: remove default
+
+    const { locationURL } = state['features/base/connection'];
+    const inBreakoutRoom = isInBreakoutRoom(state);
+    const roomId = getCurrentRoomId(state);
+    const room = md5.hex(`${locationURL?.href}${inBreakoutRoom ? `|${roomId}` : ''}`);
+
+    return appendURLParam(collabServerBaseUrl, 'room', room);
+};
+
+/**
+ * Whether the whiteboard is pinned.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {boolean}
+ */
+export const isWhiteboardPinned = (state: any): boolean =>
+    getPinnedParticipant(state)?.id === WHITEBOARD_ID
+    || _electParticipantInLargeVideo(state) === WHITEBOARD_ID;
+
+/**
+ * Returns the whiteboard toolbar button props.
+ *
+ * @param {Object} state - The state from the Redux store.
+ * @returns {{
+ *    action: Function,
```

Comment:
Not a fan of this. I think the button should always be show for moderators, and it will "create it" if it doesn't exist. For others it would only show up once a moderator has created it because the data was propagated through the prosody module.

Reply:
Until I get to use it, I'm not sure I'll be able to understand properly how it works without an example. That aside, the button should still be toggleable, so we need to determine which action to use based on what we have in stage, right?

Reply:
OK, let's table this until we have the LJM room config part done.